src/Tracing/Span.php

```diff
@@ -375,26 +386,57 @@ public function finish(?float $endTimestamp = null): ?EventId
     }
 
     /**
-     * Creates a new `Span` while setting the current `Span.id` as `parentSpanId`.
+     * Creates a new {@see Span} while setting the current ID as `parentSpanId`.
      * Also the `sampled` decision will be inherited.
      *
-     * @param SpanContext $context The Context of the child span
-     *
-     * @return Span Instance of the newly created Span
+     * @param SpanContext $context The context of the child span
      */
     public function startChild(SpanContext $context): self
     {
-        $context->sampled = $this->sampled;
-        $context->parentSpanId = $this->spanId;
-        $context->traceId = $this->traceId;
+        $context = clone $context;
+        $context->setSampled($this->sampled);
+        $context->setParentSpanId($this->spanId);
+        $context->setTraceId($this->traceId);
 
         $span = new self($context);
-
         $span->spanRecorder = $this->spanRecorder;
+
         if (null != $span->spanRecorder) {
             $span->spanRecorder->add($span);
         }
 
         return $span;
     }
+
+    /**
+     * Gets the span recorder attached to this span.
+     *
+     * @internal
+     */
+    public function getSpanRecorder(): ?SpanRecorder
+    {
+        return $this->spanRecorder;
+    }
+
+    /**
+     * Detaches the span recorder from this instance.
+     */
+    public function detachSpanRecorder(): void
+    {
+        $this->spanRecorder = null;
+    }
+
+    /**
+     * Returns a string that can be used for the `sentry-trace` header.
+     */
+    public function toTraceparent(): string
+    {
+        $sampled = '';
+
+        if (null !== $this->sampled) {
+            $sampled = $this->sampled ? '-1' : '-0';
+        }
+
+        return sprintf('%s-%s%s', (string) $this->traceId, (string) $this->spanId, $sampled);
+    }
```

Comment:
This produces a different result from [the Python implementation](https://github.com/getsentry/sentry-python/blob/f6f3525f8812f60911573a8b7f71807ecf2e1052/sentry_sdk/tracing.py#L298-L305) when the sampling decision has not been made (trailing `-`):

```py
    def to_traceparent(self):
        # type: () -> str
        sampled = ""
        if self.sampled is True:
            sampled = "1"
        if self.sampled is False:
            sampled = "0"
        return "%s-%s-%s" % (self.trace_id, self.span_id, sampled)
```

@HazAT we need to define what is the correct behavior and document in https://github.com/getsentry/develop/pull/147

Reply:
I think we need to fix this in Python and document it properly.